pkg/endpoint/policy.go

```diff
@@ -210,7 +210,17 @@ func (e *Endpoint) determineAllowLocalhost(desiredPolicyKeys map[policymap.Polic
 			Identity:         identityPkg.ReservedIdentityHost.Uint32(),
 			TrafficDirection: policymap.Ingress.Uint8(),
 		}
+
+		// Remove eventual existing ingress L4 localhost restrictions
+		for key := range desiredPolicyKeys {
+			if key.Identity == identityPkg.ReservedIdentityHost.Uint32() &&
```

Comment:
This seems fine for now. I'm somewhat confused as to how this regression got introduced, though. Is it because the L3-only entries for localhost were not getting accessed in the PolicyMap?